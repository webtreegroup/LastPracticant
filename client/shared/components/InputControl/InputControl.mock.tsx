import React, { FC, memo } from 'react';
import { useForm } from 'react-hook-form';
import { InputControl } from './InputControl';

export const InputControlMock: FC = memo(() => {
    const { control } = useForm();
    return (
        <InputControl
            variant="outlined"
            name="login"
            label="Login"
            control={control}
        />
    );
});
