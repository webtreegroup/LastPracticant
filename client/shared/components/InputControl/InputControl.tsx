import React from 'react';
import { Control, Controller, RegisterOptions } from 'react-hook-form';
import { TextField, OutlinedTextFieldProps } from '@material-ui/core';
import { PatternProps, CHECK_REQUIRED } from 'client/shared/consts';

export interface InputDataProps {
    name: string;
    label: string;
    control?: Control;
    pattern?: PatternProps;
    type?: string;
    required?: boolean;
    multiline?: boolean;
    rows?: number
    rowsMax?: number
}

type InputControlProps = InputDataProps & OutlinedTextFieldProps;

export const InputControl: React.FC<InputControlProps> = React.memo((props) => {
    const {
        name,
        control,
        type = 'text',
        required,
        pattern,
    } = props;
    const rules: RegisterOptions = {
        required: required && CHECK_REQUIRED,
        pattern,
    };
    return (
        <Controller
            name={name}
            control={control}
            defaultValue=""
            rules={rules}
            render={({ onChange, value }) => (
                <TextField
                    type={type}
                    onChange={onChange}
                    value={value || ''}
                    {...props}
                />
            )}
        />
    );
});
