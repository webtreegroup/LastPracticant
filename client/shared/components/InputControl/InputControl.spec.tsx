import React from 'react';
import {
    cleanup, fireEvent, render, screen,
} from '@testing-library/react';
import { InputControlMock } from './InputControl.mock';

afterEach(cleanup);

describe('InputControl', () => {
    it('Render InputControl component', () => {
        const { asFragment } = render(<InputControlMock />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Render InputControl component onchange value', () => {
        const { asFragment } = render(<InputControlMock />);
        fireEvent.change(screen.getByRole('textbox'), {
            target: { value: 'Set new value' },
        });
        expect(asFragment()).toMatchSnapshot();
    });
});
