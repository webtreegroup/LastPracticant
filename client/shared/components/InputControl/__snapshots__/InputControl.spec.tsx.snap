// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControl Render InputControl component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
    control="[object Object]"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      Login
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        name="login"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-3"
        >
          <span>
            Login
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InputControl Render InputControl component onchange value 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
    control="[object Object]"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink="true"
    >
      Login
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        name="login"
        type="text"
        value="Set new value"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
        >
          <span>
            Login
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
